CONTRIBUTING.md

```diff
@@ -14,22 +14,44 @@ Please note that this project is released with a [Contributor Code of Conduct][c
 
 - Follow the style guide for [Powershell](https://github.com/PoshCode/PowerShellPracticeAndStyle) when writing Windows scripts. There is currently no set style for the Shell scripts that run Linux installs :soon:.
-- Include complete details of why this is needed in the PR description. If it's a new piece tool being installed, consider cross-platform. If the tool is available in other platforms (MacOS, Windows, Linux), make sure you include it in as many as possible.
+- Include complete details of why this is needed in the PR description. 
 - Keep your change as focused as possible. If there are multiple changes you would like to make that are not dependent upon each other, consider submitting them as separate pull requests.
 - Write [good commit messages](http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
+- For new tools:
+  - Make sure that the tool satisfies [Software Guidelines](README.md#software-guidelines).
+  - Create an issue and get an approval from us to add this tool to the image before creating the pull request.
+
+## How to add new tool
+### General rules
+- For every new tool add validation scripts and update software report script to make sure that it is included to documentation
+- If the tool is available in other platforms (MacOS, Windows, Linux), make sure you include it in as many as possible.
+- If tools installation includes a few versions, consider putting the list of versions to `toolset.json` file. It will help other customers to configure their builds flexibly. See [toolset-windows-2016.json](images/win/toolsets/toolset-2019.json) as example.
+- Use consistent naming across all files
+- Validation scripts should be simple and shouldn't change image content
+
+### Windows
+- Add script that will install tool and put it to `scripts/Installers` folder.  
+There are a bunch of helper functions that could simplify your code: `Choco-Install`, `Install-Binary`, `Install-VsixExtension`, ` Start-DownloadWithRetry`, `Test-IsWin16`, ` Test-IsWin19` (Find the full list of helpers in [ImageHelpers.psm1](images/win/scripts/ImageHelpers/ImageHelpers.psm1)).
+- Add script that will validate tool installation and put it to `scripts/Tests` folder.  
+We use [Pester v5](https://github.com/pester/pester) for validation scripts. Create the separate `*.Tests.ps1` file in case if tool requires complex tests or use `Tools.Tests.ps1` for simple tests.  
+Run your test by adding `Invoke-PesterTests -TestFile <testFileName> [-TestName <describeName>]` at the end of installation script to make sure that your tests will be run.
+- Add changes to document tool name and version to the software report generator `images/win/scripts/SoftwareReport/SoftwareReport.Generator.ps1` that is used to generate software README file, e.g. [Windows2019-Readme.md](images/win/Windows2019-Readme.md). We use [MarkdownPS](https://github.com/Sarafian/MarkdownPS) to build documentation for image.
```

Comment:
I would swap parts of the sentence: `Add changes to the software README report generator `images/win/scripts/SoftwareReport/SoftwareReport.Generator.ps1` to document tool name and version, e.g. [Windows2019-Readme.md](images/win/Windows2019-Readme.md). `